WrightTools/data/_join.py

```diff
@@ -47,13 +71,20 @@ def join(datas, *, name="join", parent=None, verbose=True) -> Data:
         A new Data instance.
     """
     warnings.warn("join", category=wt_exceptions.EntireDatasetInMemoryWarning)
-    # TODO: fill value
+    if isinstance(datas, Collection):
+        datas = datas.values()
     datas = list(datas)
+    if not isinstance(atol, collections.Iterable):
```

Comment:
consider `hasattr(atol, '__iter__')`

Reply:
https://docs.python.org/3/library/collections.abc.html#collections.abc.Iterable

This check subsumes the `hasattr` check, though this does raise the point that the Iterable class has moved to `collections.abc` rather than just `collections` (though it is still available there in current python, I think it may be slated for removal in 3.8)

Reply:
the other option is to try/except `iter(atol)`